command/acl_auth_method_update.go

```diff
@@ -0,0 +1,191 @@
+package command
+
+import (
+	"flag"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/nomad/api"
+	"github.com/mitchellh/cli"
+	"github.com/posener/complete"
+	"golang.org/x/exp/slices"
+)
+
+// Ensure ACLAuthMethodUpdateCommand satisfies the cli.Command interface.
+var _ cli.Command = &ACLAuthMethodUpdateCommand{}
+
+// ACLRoleUpdateCommand implements cli.Command.
+type ACLAuthMethodUpdateCommand struct {
+	Meta
+
+	methodType    string
+	tokenLocality string
+	maxTokenTTL   time.Duration
+	isDefault     bool
+	config        string
+}
+
+// Help satisfies the cli.Command Help function.
+func (a *ACLAuthMethodUpdateCommand) Help() string {
+	helpText := `
+Usage: nomad acl auth-method update [options] <acl_auth_method_name>
+
+  Update is used to update ACL auth methods. Use requires a management token.
+
+General Options:
+
+  ` + generalOptionsUsage(usageOptsDefault|usageOptsNoNamespace) + `
+
+ACL Auth Method Update Options:
+
+  -type
+    Updates the type of the auth method. Currently the only supported type is
+    'OIDC'.
+
+  -max-token-ttl
+    Updates the duration of time all tokens created by this auth method should be
+    valid for.
+
+  -token-locality
+    Updates the kind of token that this auth method should produce. This can be
+    either 'local' or 'global'. If empty the value of 'local' is assumed.
+
+  -default
+    Specifies whether this auth method should be treated as a default one in
+    case no auth method is explicitly specified for a login command.
+
+  -config
+    Updates auth method configuration (in JSON format).
+`
+
+	return strings.TrimSpace(helpText)
+}
+
+func (a *ACLAuthMethodUpdateCommand) AutocompleteFlags() complete.Flags {
+	return mergeAutocompleteFlags(a.Meta.AutocompleteFlags(FlagSetClient),
+		complete.Flags{
+			"-type":           complete.PredictAnything,
+			"-max-token-ttl":  complete.PredictAnything,
+			"-token-locality": complete.PredictAnything,
+			"-default":        complete.PredictAnything,
```

Comment:
We could use complete.PredictSet("true", "false") here as we will always be looking for known values.